raiden/raiden_service.py

```diff
@@ -539,27 +541,39 @@ def handle_secret(  # pylint: disable=too-many-arguments
                     )
 
                     if is_balance_proof:
-                        channel.register_transfer(partner_secret_message)
+                        channel.register_transfer(
+                            self.get_block_number(),
+                            partner_secret_message,
+                        )
                         channels_to_remove.append(channel)
                     else:
                         channel.register_secret(secret)
-                        self.send_async(
+                        messages_to_send.append((
                             channel.partner_state.address,
```

Comment:
It's probably an overkill only for these 2 places here, but this could be a named tuple, instead of a simple tuple.

Reply:
Well, for this instance I think it's overkill, this tuple is not exposed outside the method.

Reply:
yep I agree. My OCD nature compelled me to mention it.